api/tests/acceptance/application/assessments/assessment-controller-auto-validate-next-challenge_test.js

```diff
@@ -0,0 +1,104 @@
+const {
+  expect,
+  databaseBuilder,
+  knex,
+  mockLearningContent,
+  learningContentBuilder,
+  insertUserWithRoleSuperAdmin,
+  generateValidRequestAuthorizationHeader,
+} = require('../../../test-helper');
+const settings = require('../../../../lib/config');
+const createServer = require('../../../../server');
+const Assessment = require('../../../../lib/domain/models/Assessment');
+
+const lastChallengeAnswer = 'last challenge answer';
+const lastChallengeId = 'lastChallengeId';
+const learningContent = [
+  {
+    areas: [
+      {
+        id: 'recArea1',
+        titleFrFr: 'area1_Title',
+        color: 'someColor',
+        competences: [
+          {
+            id: 'competenceId',
+            nameFrFr: 'Mener une recherche et une veille d’information',
+            index: '1.1',
+            tubes: [
+              {
+                id: 'recTube0_0',
+                skills: [
+                  {
+                    id: 'skillWeb2Id',
+                    nom: '@web2',
+                    challenges: [
+                      { id: lastChallengeId, solution: lastChallengeAnswer, type: 'QROC', autoReply: false },
+                    ],
+                  },
+                ],
+              },
+            ],
+          },
+        ],
+      },
+    ],
+  },
+];
+
+describe('Acceptance | API | assessment-controller-auto-validate-next-challenge', function () {
+  let originalEnvValue;
+  let server;
+  let assessmentId;
+
+  beforeEach(async function () {
+    originalEnvValue = settings.featureToggles.isAlwaysOkValidateNextChallengeEndpointEnabled;
+    settings.featureToggles.isAlwaysOkValidateNextChallengeEndpointEnabled = true;
+
+    server = await createServer();
+    const learningContentObjects = learningContentBuilder.buildLearningContent(learningContent);
+    mockLearningContent(learningContentObjects);
+  });
+
+  afterEach(function () {
+    settings.featureToggles.isAlwaysOkValidateNextChallengeEndpointEnabled = originalEnvValue;
+  });
+
+  describe('POST /api/admin/assessments/:id/always-ok-validate-next-challenge', function () {
+    let userId;
+
+    beforeEach(async function () {
+      const user = await insertUserWithRoleSuperAdmin();
+      userId = user.id;
+      assessmentId = databaseBuilder.factory.buildAssessment({
+        state: Assessment.states.STARTED,
+        type: Assessment.types.PREVIEW,
+        lastChallengeId,
+        userId,
+      }).id;
+      await databaseBuilder.commit();
+    });
+
+    afterEach(async function () {
+      await knex('answers').delete();
+    });
+
+    it('should return 200 HTTP status code', async function () {
```

Comment:
**nitpick**
Il fait un peu plus que ce qu'il dit
```suggestion
    it('should return 200 HTTP status code and get a valid answers', async function () {
```

Reply:
@octo-topi tu as tout à fait raison sur le fait que ce `it` devrait être plus précisément décrit. Mais je mettrais plutôt la formulation suivante alors :

`'records an "ok" answer and returns 200 HTTP status code'`

Qu'en penses-tu ?

Reply:
Oui, j'ai mis `get` au sens de `obtient` uen réponse valide (c'est pas juste un save, c'est aussi une évaluation juste/faux). 
Mais dans le contexte IT, `get` c'est de la lecture donc c'est `misledading`.

Reply:
Voilà, la description du test a été mise à jour.